projects/plugins/backup/src/js/tools/jetpack-rest-api-client/index.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+
+/**
+ * Helps create new custom error classes to better notify upper layers.
+ *
+ * @param {string} name - The Error name that will be available in `Error.name`.
+ * @returns {Error}      a new custom error class.
+ */
+function createCustomError( name ) {
+	class CustomError extends Error {
+		constructor( ...args ) {
+			super( ...args );
+			this.name = name;
+		}
+	}
+	return CustomError;
+}
+
+export const JsonParseError = createCustomError( 'JsonParseError' );
+export const JsonParseAfterRedirectError = createCustomError( 'JsonParseAfterRedirectError' );
+export const Api404Error = createCustomError( 'Api404Error' );
+export const Api404AfterRedirectError = createCustomError( 'Api404AfterRedirectError' );
+export const FetchNetworkError = createCustomError( 'FetchNetworkError' );
+
+/**
+ * Jetpack REST API Client.
+ *
+ * @param {string} root - The API root URL.
+ * @param {string} nonce - The API nonce.
+ * @class
+ */
+function JetpackRestApiClient( root, nonce ) {
```

Comment:
Would it make sense to try to rely on `@wordpress/api-fetch` and its `apiFetch` for some of this?

Reply:
I was using what Connect was doing as the example here.  It may have just been using the Jetpack Plugin as an example.  I only do each fetch in one place, so if I can use api-fetch in-line and make this file obsolete, I'm all for that.

Reply:
Yeah, it does look like it followed the Jetpack dashboard's example, which was built before we had WordPress JS packages. I think it would make sense to simplify and rely on WordPress' work when possible.